ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/AbstractArcticOptimizePlan.java

```diff
@@ -84,12 +82,16 @@ public AbstractArcticOptimizePlan(ArcticTable arcticTable, TableOptimizeRuntime
                                     List<ContentFileWithSequence<?>> changeFiles,
                                 List<FileScanTask> baseFileScanTasks,
                                 int queueId, long currentTime, long changeSnapshotId, long baseSnapshotId) {
-    super(arcticTable, tableOptimizeRuntime, queueId, currentTime);
+    super(arcticTable, tableOptimizeRuntime, queueId, currentTime, baseSnapshotId);
     this.baseFileScanTasks = baseFileScanTasks;
     this.changeFiles = changeFiles;
     this.isCustomizeDir = false;
     this.currentChangeSnapshotId = changeSnapshotId;
-    this.currentBaseSnapshotId = baseSnapshotId;
+  }
+
+  @Override
+  protected boolean limitFileCnt() {
+    return false;
```

Comment:
Why do the mixed format tables optimizing not limit file count?

Reply:
AFAIK, the partition weight for mixed format table optimizing may not be a simple number, am I right?
If yes, the current `PartitionWeight` definition may not be good for scaling.

Reply:
We can support mixed-format tables for this feature in another PR, but we may create an issue to address that.

Reply:
OK, I create another issue to address the mix-format table. And I will abstract an interface `Weight` to replace a simple long.

Reply:
related issue https://github.com/NetEase/arctic/issues/1213